src/plugins/intel_gpu/src/runtime/ocl/ocl_device_detector.cpp

```diff
@@ -91,47 +92,58 @@ static std::vector<cl::Device> getSubDevices(cl::Device& rootDevice) {
     return subDevices;
 }
 
+std::vector<device::ptr> ocl_device_detector::sort_devices(const std::vector<device::ptr>& devices_list) {
+    std::vector<device::ptr> sorted_list;
+    size_t insert_idx = 0;
+    for (auto& dptr : devices_list) {
+        if (dptr->get_info().dev_type == cldnn::device_type::integrated_gpu) {
```

Comment:
Very theoretical situation, but anyway.. Shouldn't we make sure that it is Intel device as well before moving it at first position?

Reply:
changed